juriscraper/opinions/united_states/federal_bankruptcy/bap1.py

```diff
@@ -0,0 +1,124 @@
+        "Bankruptcy Court of MA - Boston": "MB",
+        "Bankruptcy Court of MA - Springfield": "MS",
+        "Bankruptcy Court of ME - Portland": "EP",
+        "Bankruptcy Court - Rhode Island": "RI",
+        "Bankruptcy Court - San Juan Puerto Rico": "PR",
+        "Bankruptcy Court of MA - Worcester": "MW",
+        "Bankruptcy Court - Ponce Puerto Rico": "PR",
+        "Bankruptcy Court of NH, Concord": "NH",
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.base_url = self.url = "https://www.bap1.uscourts.gov/bapopn"
+        self.court_id = self.__module__
+
+        self.set_date_filter_parameters()
+
+        # There are 29 pages as of development in Dec 2023 (source indexes from 0)
+        self.back_scrape_iterable = range(29)
+
+    def _download(self, request_dict: Dict = {}) -> HtmlElement:
+        return super()._download(self.parameters)
+
+    def _process_html(self) -> None:
+        for row in self.html.xpath("//tr[td]"):
+            opinion_name = row.xpath("td[1]")[0].text_content().strip()
+            status = self.get_status_from_opinion_name(opinion_name)
+
+            lower_court = row.xpath("td[4]/span")[0].text_content().strip()
+            docket_number = row.xpath("td[2]")[0].text_content().strip()
+
+            case = {
+                "status": status,
+                "url": row.xpath("td[1]/a/@href")[0],  # opinion url
+                "docket": self.build_full_docket_number(
+                    docket_number, lower_court
+                ),
+                # Pub Date
+                "date": row.xpath("td[3]")[0].text_content().strip(),
+                # short title
+                "name": row.xpath("td[4]")[0].text.strip(),
+                # district
+                "lower_court": lower_court,
+            }
+
+            self.cases.append(case)
+
+    def set_date_filter_parameters(self):
+        """
+        Limit search to get fresh data and avoid overloading the court server
+        Default: last month
+        """
+        date_filter_end = datetime.today()
+        date_filter_start = date_filter_end - timedelta(30)
+        self.parameters = {
+            "params": {
+                "opn": "",
+                "field_opn_short_title_value": "",
+                "field_opn_issdate_value[min][date]": date_filter_start.strftime(
+                    "%m/%d/%Y"
+                ),
+                "field_opn_issdate_value[max][date]": date_filter_end.strftime(
+                    "%m/%d/%Y"
+                ),
+            }
+        }
+
+    def _download_backwards(self, page_number: int) -> None:
+        self.parameters = {}  # delete date filters used in normal scraper
+        self.url = (
+            f"{self.base_url}?page={page_number}"
+            if page_number > 0
+            else self.base_url
+        )
+        self.html = self._download()
+        self._process_html()
```

Comment:
You can call this here but you dont need to.  in this case the parse method would run this code
```
    def parse(self):
        if not self.downloader_executed:
            # Run the downloader if it hasn't been run already
            self.html = self._download()

            # Process the available html (optional)
            self._process_html()
            
            # and so on
```